### PR TITLE
table of oidc vs oauth2 endpoints

### PR DESCRIPTION
**Pull request by mystiberry-okta**
@cwu-okta and @federations-okta  team, Here's the table Cindy suggested showing OIDC and OAuth 2.0 endpoints. One problem in the last two rows, I can't link the normal way (markdown doesn't parse [{ correctly), I hope to figure it out tomorrow.

Meanwhile, please let me know if this is what people wanted and if you see any issues. Thanks!

### CODE REVIEW

File: _source/_docs/api/resources/oauth2.md [hunk 1]
```diff
@@ -272,6 +272,20 @@ The actions in a Rule define which scopes can be granted to the requests, thus d
...
+|  [Token](#token-request) | {% api_operation post /oauth2/v1/token %} {% api_lifecycle ea %}               | {% api_operation post /oauth2/:authorizationServerId/v1/token %} {% api_lifecycle beta %}          |
+|  [Introspection Request](#introspection-request)  | {% api_operation post /oauth2/v1/introspect %} {% api_lifecycle ea %}               | {% api_operation post /oauth2/:authorizationServerId/v1/introspect %} {% api_lifecycle beta %}          |
+|  [Revocation Request](#revocation-request)  | {% api_operation post /oauth2/v1/revoke %} {% api_lifecycle ea %}               | {% api_operation post /oauth2/:authorizationServerId/v1/revoke %} {% api_lifecycle beta %}          |
+|  Get Keys  | {% api_operation get /oauth2/v1/keys %} (% api_lifecycle ea %}               |  {% api_operation get /oauth2/:authorizationServerId/v1/keys %} {% api_lifecycle beta %}         |
+|  Authorization Server Metadata Well-Known Discovery Document  | {% api_operation get /.well-known/openid-configuration %} {api_lifecycle ea %}               | {% api_operation get /oauth2/:authorizationServerId/.well-known/oauth-authorization-server %} {% api_lifecycle beta %}          |
```
Comment (larsjohansen-okta) [inline _source/_docs/api/resources/oauth2.md]: We have a well-known discovery document for OpenID Connect at ```/oauth2/:authorizationServerId/.well-known/openid-configuration ``` as well, see my overall comment for details.

**Comment by larsjohansen-okta** [comment]
This is good! I still think it's weird that we're not more explicit about the fact that you can use OAuth 2.0 endpoints for OpenID Connect as well. This table makes it look like you can only do OpenID Connect for the specified endpoints.

**Comment by cwu-okta** [comment]
I did not mean to create a table; instead, I meant to weave the content into oauth2.md itself.

For example we have this 
image
I can't tell which is OIDC and which is OAuth. I would hope for something like

```
Token Request

OpenID Connect: /oauth2/v1/token [early access liquid tag]
OAuth: /oauth2/:authorizationServerId/v1/token [beta liquid tag]
```

At the top of oauth2 I would also like some introductory text explaining that OAuth and OIDC are very similar and one should follow the OIDC endpoints if they are following the OIDC use case and OAuth if they are following the OAuth use case. I suggest following up with Darko if you need help on the intro paragraph(s)

@tthompson-okta @yuliu-okta  fyi

**Comment by mystiberry-okta** [comment]
@cwu-okta Got it, easy to fix--but what about Lars' point that you can use OAuth 2.0 endpoints for OIDC? Darko's working on the intro as we speak.

File: _source/_docs/api/resources/oauth2.md (hunk 1 above)
Comment (mystiberry-okta) [inline _source/_docs/api/resources/oauth2.md]: Got it!

Comment (mystiberry-okta) [inline _source/_docs/api/resources/oauth2.md]: @larsjohansen-okta in oauth2.md, there is only /oauth2/:authorizationServerId/.well-known/oauth-authorization-server, and in oidc, there is only ./well-known/openid-configuration .  We haven't doc'd /oauth2/:authorizationServerId/.well-known/openid-configuration

??

**Comment by mystiberry-okta** [comment]
@cwu-okta and @larsjohansen-okta Here are my updates. I don't have answers to Lars question (shouldn't we point out that you can use OAuth 2.0 endpoints for OIDC?) and no sign of the mystery endpoint Lars said exists: 

We have a well-known discovery document for OpenID Connect at /oauth2/:authorizationServerId/.well-known/openid-configuration as well, see my overall comment for details.

File: _source/_docs/api/resources/oauth2.md [hunk 2]
```diff
@@ -776,7 +792,9 @@ Standard open-source libraries are available for every major language to perform
 ### Authorization Server Metadata
 {:.api .api-operation}
 
-{% api_operation get /oauth2/:authorizationServerId/.well-known/oauth-authorization-server %} {% api_lifecycle beta %}
+OpenID Connect: {% api_opertion get /.well-known/openid-configuration %} {% api_lifecycle ea %}
+
+Oauth 2.0: {% api_operation get /oauth2/:authorizationServerId/.well-known/oauth-authorization-server %} {% api_lifecycle beta %}
```
Comment (larsjohansen-okta) [inline _source/_docs/api/resources/oauth2.md]: Oauth -> OAuth

File: _source/_docs/api/resources/oauth2.md [hunk 3]
```diff
@@ -653,7 +662,11 @@ Content-Type: application/json;charset=UTF-8
 ### Revocation Request
 {:.api .api-operation}
 
+{% beta %}
+OpenID Connect: 
+{% endbeta %}
 {% api_operation post /oauth2/v1/revoke %} {% api_lifecycle ea %}
+
 {% beta %}
 {% api_operation post /oauth2/:authorizationServerId/v1/revoke %} {% api_lifecycle beta %}
```
Comment (larsjohansen-okta) [inline _source/_docs/api/resources/oauth2.md]: Shouldn't we have ```OAuth 2.0:``` within this beta section too?

Comment (mystiberry-okta) [inline _source/_docs/api/resources/oauth2.md]: Fixed!

**Comment by mystiberry-okta** [comment]
Since Cindy is gone, can anyone in @federations-okta  answer these two questions, so I can ship this?

1. I don't have answers to Lars question (shouldn't we point out that you can use OAuth 2.0 endpoints for OIDC?) 

2. I can't find the mystery endpoint Lars said exists:

"We have a well-known discovery document for OpenID Connect at /oauth2/:authorizationServerId/.well-known/openid-configuration as well, see my overall comment for details."

File: _source/_docs/api/resources/oauth2.md [hunk 4]
```diff
@@ -282,8 +282,7 @@ OpenID Connect:
 
 {% beta %}
 
-OAuth 2.0:
-
+OAuth 2.0:
```
Comment (yuliu-okta) [inline _source/_docs/api/resources/oauth2.md]: @mystiberry-okta As the overview section says, OIDC, API Access Management and Social Auth are features built on top of OAuth 2.0. So I'd suggest we change the tag from OAuth 2.0 to API Access Management, it is really that feature is Beta, not all OAuth 2.0 API. And also we can add Social Auth in the authorization endpoint.

Comment (mystiberry-okta) [inline _source/_docs/api/resources/oauth2.md]: I wish we had an Okta feature name for OIDC to make the naming truly parallel. Like this suggestion. I'll fix these tags, but wondering if we should make this change everywhere in our docs?

**Comment by mcguinness** [comment]
I think is future revision it may be beneficial to break out the Organization Authorization Server and Custom Authorization Server.  OIDC only lives on the Org Authorization Server.   We don't support custom scopes for the Okta API be when we do they will be available on the Org Authorization Server.  Its confusing to see multiple endpoints that appear to do the same if you don't grok the different between Org and Custom Authorization Servers.    I think it may also be beneficial to introduce the term security domain and show the relationship of Resources Servers and Authorization Server as security domain.  The Org AS is bound by the org security domain and tokens issued by the Org AS should only be used by the client (id_token) or an protected resource for the Okta API where as Custom AS security domain is defined by the customer.

**Comment by mystiberry-okta** [comment]
I'll create a JIRA for Karl's comments and role back the changes I made (except for the typo fixes) as I don't believe this is the right solution for the issue Cindy raised. Also a JIRA to fix the OAuth 2.0 vs. API Access Management. @federations-okta

**Comment by mystiberry-okta** [comment]
Consensus is that this approach doesn't work. Created new JIRA to get this all worked out: https://oktainc.atlassian.net/browse/OKTA-107474

**Pull request closed without merging**